solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -599,6 +602,28 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
             for (SearchComponent c : components) {
               c.handleResponses(rb, srsp.getShardRequest());
             }
+
+            // Compute total CpuTime used by all shards.
+            List<ShardResponse> responses = srsp.getShardRequest().responses;
+            for (ShardResponse response : responses) {
+              if ((response.getSolrResponse() != null)
+                  && (response.getSolrResponse().getResponse() != null)
+                  && (response.getSolrResponse().getResponse().get("responseHeader") != null)) {
+                @SuppressWarnings("unchecked")
+                SimpleOrderedMap<Object> header =
+                    (SimpleOrderedMap<Object>)
+                        response
+                            .getSolrResponse()
+                            .getResponse()
+                            .get(SolrQueryResponse.RESPONSE_HEADER_KEY);
+                if (header != null) {
+                  Long shardCpuTime = (Long) header.get(ThreadStats.CPU_TIME);
+                  if (shardCpuTime != null) {
+                    totalShardCpuTime += shardCpuTime;
+                  }
+                }
+              }
+            }
```

Comment:
could you extract out a method for this to try to keep handleRequestBody less long?
Gate by "publishCpuTime" as well?